network/wifi/wifi.c

```diff
@@ -94,13 +113,67 @@ static int32_t wifi_socket_recvfrom(struct wifi_desc *desc, uint32_t sock_id,
 				    void *data, uint32_t size,
 				    struct socket_address *from);
 
-/* Connect internal functions to the network interface */
-static void wifi_init_intarface(struct wifi_desc *desc)
+/* Returns the index of a socket in SOCKET_UNUSED state */
+static inline uint32_t _wifi_get_unused_socket(struct wifi_desc *desc)
+{
+	uint32_t i;
+
+	for (i = 0; i < NB_SOCKETS; i++)
+		if (desc->sockets[i].state == SOCKET_UNUSED) {
+			desc->sockets[i].state = SOCKET_DISCONNECTED;
+			return i;
+		}
+
+	return (uint32_t)-1;
```

Comment:
this could be an error code like ENOENT

Reply:
also remove magic number -1 probably add a define to use in this case.

Reply:
solved